src/djimatrice_drone.py

```diff
@@ -0,0 +1,230 @@
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/mission_waypoint_getSpeed', 'dji_sdk/MissionWpGetSpeed')
+            request = roslibpy.ServiceRequest()
+
+            print('Calling mission_waypoint_setSpeed service...')
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Failed to fetch drone speed"}
+        return result
+
+    def start_mission(self):
+        # TODO: Can start on this once waypoints are implemented
+
+        # if (self.flight_status == Drone.Flight_Status.ON_GROUND_STANDBY):
+        #     if (self.prev_flight_status == Drone.Flight_Status.NULL):
+        #         self.flight_status = Drone.Flight_Status.FLYING
+        #         self.prev_flight_status = Drone.Flight_Status.ON_GROUND_STANDBY
+        #         command_list = np.zeros(16)
+        #         command_params = np.zeros(16)
+        #         for i in range(self.waypoints_count):
+        #             waypoint = self.waypoints[i]
+        #             waypoint_coord = waypoint.position
+        #             # Assumption: position stored as dictionary
+        #             waypoint_msg = {}
+        #             #TODO: Find out where waypoints are being added
+        #     else:
+        #         self.update_mission_helper(Drone.UpdateMissionAction.CONTINUE_MISSION)
+
+        try:
+            print("Attempting to start drone mission...")
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/mission_waypoint_action', 'dji_sdk/MissionWpAction')
+            request = roslibpy.ServiceRequest({"action": Drone.WaypointActions.START})
+
+            print('Calling mission_waypoint_action start service...')
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Mission failed to start"}
+        # self.start_mission_callback(result)
+        # TODO: Upon failure, revert back to original setting
+        return result
+
+    def start_mission_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result["success"]
+        
+    def stop_mission(self):
+        try:
+            print("Attempting to stop drone mission...")
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/mission_waypoint_action', 'dji_sdk/MissionWpAction')
+            request = roslibpy.ServiceRequest({"action": Drone.WaypointActions.STOP})
+
+            print('Calling mission_waypoint_action stop service...')
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Mission failed to stop"}
+        self.stop_mission_callback(result)
+        # TODO: Upon failure, revert back to original setting
```

Comment:
what do you mean original setting?

Reply:
The previous state should be the original setting (like the saved drone state & location before this failure occurred).